erigon-lib/diagnostics/stages.go

```diff
@@ -83,6 +83,10 @@ func ReadStagesList(db kv.RoDB) []string {
 	var info []string
 	err := json.Unmarshal(data, &info)
 
+	if len(data) == 0 {
+		return []string{}
```

Comment:
you can do that checks straight after you fetched `data` which is preferable way. So, you got data, you validate if its empty, if not - call `Unmarshal` and check `err` after call